### PR TITLE
Add sys.flags

### PR DESCRIPTION
**Pull request by skinny121**
For #958. For this, I created a `pystruct_sequence` proc macro as there are quite a few of these 'struct sequences' throughout the c part of the standard library.

### CODE REVIEW

File: derive/src/pyclass.rs [hunk 1]
```diff
@@ -316,11 +318,111 @@ pub fn impl_pyclass(attr: AttributeArgs, item: Item) -> Result<TokenStream2, Dia
...
+                };
+                methods.push(method);
+                let field_name_str = field_name.to_string();
+                // TODO add doc to the generated property
+                let method_reference = quote! {
+                    class.set_str_attr(
+                        #field_name_str,
+                        ::rustpython_vm::obj::objproperty::PropertyBuilder::new(ctx)
+                            .add_getter(Self::#method_ident)
+                            .create(),
```
Comment (coolreader18) [inline derive/src/pyclass.rs]: You probably don't need to add it as a method that gets put as an item on the struct, you should be able to generate code like:

```rust
class.set_str_attr(
    "#field_name_str",
    ::rustpython_vm::obj::objproperty::PropertyBuilder::new(ctx)
        .add_getter({
            fn getter(zelf: ..., _vm: ...) -> ... { ... }
            getter
        })
        .create(),
);
```

File: derive/src/pyclass.rs [hunk 2]
```diff
@@ -316,11 +318,111 @@ pub fn impl_pyclass(attr: AttributeArgs, item: Item) -> Result<TokenStream2, Dia
...
+                }
+            }
+        };
+        Ok(ret)
+    } else {
+        bail_span!(
+            item,
+            "#[pystruct_sequence] can only be on a struct declaration"
+        )
+    }
```
Comment (coolreader18) [inline derive/src/pyclass.rs]: If you want, I'd say you could put `let struc = if let ... { struc } else { bail_span!(...) };` at the beginning, to help keep the bulk of the code less indented.

File: derive/src/pyclass.rs (hunk 1 above)
Comment (skinny121) [reply derive/src/pyclass.rs]: I was able to use a closure here, which is quite nice.

File: derive/src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#![recursion_limit = "128"]
```
Comment (windelbouwman) [inline derive/src/lib.rs]: Why is this required? I have my doubts about changing the recursion limit to this large value. Is there a better way to implement the macros?

Comment (coolreader18) [reply derive/src/lib.rs]: This is just for the quote macros, it doesn't have any effect on macros in `rustpython_vm`.

Comment (windelbouwman) [reply derive/src/lib.rs]: Okay, its probably fine. The macro derive package is still a bit voodoo to me :)

**Comment by coolreader18** [comment]
Merging.

**Comment by skinny121** [comment]
Done.
